src/scripts/common.js

```diff
@@ -512,66 +502,58 @@ window.togglbutton = {
     // Add created link to links array
     togglbutton.links.push({ params: params, link: link });
 
-    // new button created - set state
+    return link;
+  },
+
+  // Query active timer entry, and set it to active. If a callback is
+  // provided as an argument, it will be invoked before updating the
+  // timer links.
+  queryAndUpdateTimerLink: function () {
     chrome.runtime.sendMessage({ type: 'currentEntry' }, function (response) {
-      let currentEntry; let i;
-      if (response.success) {
-        currentEntry = response.currentEntry;
-        for (i = 0; i < togglbutton.links.length; i++) {
-          link = togglbutton.links[i].link;
-          if (
-            invokeIfFunction(togglbutton.links[i].params.description) ===
-            currentEntry.description
-          ) {
-            activate();
-          } else {
-            deactivate();
-          }
-        }
-      }
+      togglbutton.updateTimerLink(response.currentEntry);
     });
-
-    return link;
   },
 
-  // If "entry" is passed, make button active; otherwise inactive.
+  // Make button corresponding to 'entry' active, if any. ; otherwise inactive.
   updateTimerLink: function (entry) {
-    let linkText = '';
-
-    let color = '';
-
-    let link;
-
-    let i;
-
-    let minimal;
+    if (entry) {
+      for (const current of togglbutton.links) {
+        if (invokeIfFunction(current.params.description) !== entry.description) {
+          togglbutton.deactivateTimerLink(current.link);
```

Comment:
Is this required, since `activateTimerLink` at L524 will deactivate all other links anyway?

Reply:
If not, we could make it simpler by replacing L520-L529 with

```javascript
const current = togglbutton.links.find(button => invokeIfFunction(button.params.description) === entry.description);
togglbutton.activateTimerLink(current.link);
```

Reply:
Adding in here that we've seen at least one integration in testing that now gets stuck in an infinite re-rendering loop (Atlassian Jira Cloud, issue page) when the button becomes "active". Wondering if it's related to the above *(I have not debugged it fully myself at this time, sorry)*.

Reply:
:+1: Modified it to avoid the redundant deactivations and use `find`